rust/crd/src/lib.rs

```diff
@@ -88,6 +89,9 @@ pub struct NifiSpec {
     /// Emergency stop button, if `true` then all pods are stopped without affecting configuration (as setting `replicas` to `0` would)
     #[serde(default, skip_serializing_if = "Option::is_none")]
     pub stopped: Option<bool>,
+    /// Extra volumes to mount into every container, this can be useful to for example make client
+    /// certificates, keytabs or similar things available to processors
+    pub extra_volumes: Option<Vec<Volume>>,
```

Comment:
I don't think there's a difference here between `None` and `Some(vec![])`, right?

Reply:
I'm never really sure about that distinction and what to use tbh..

Reply:
Is `None' a distinct value that the user might specify (for example, to disable some given feature)? Wrap it in an option.

Is the value atomic, so that the user needs a way to specify "no, I don't want to override this" value? Wrap it in an option.

In this case neither applies. If you don't want to specify any extra volumes, supply an empty list. We could merge either by name or by just concatenating the lists... in either case an empty list would be a no-op.

Reply:
Would ommitting the option mean that users have to specify this even if they do not want extra options? 
Or would we simply serde default this to an empty list for that case?

Reply:
We should be fine just serde-defaulting it.

Reply:
should be addressed now